src/txn/txn_recover.c

```diff
@@ -452,6 +452,69 @@ __recovery_set_oldest_timestamp(WT_RECOVERY *r)
     return (0);
 }
 
+/*
+ * __recovery_set_checkpoint_snapshot --
+ *     Set the checkpoint snapshot details as retrieved from the metadata file.
+ */
+static int
+__recovery_set_checkpoint_snapshot(WT_RECOVERY *r)
+{
+    WT_CONFIG list;
+    WT_CONFIG_ITEM cval;
+    WT_CONFIG_ITEM k;
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+    uint8_t counter;
+    char *sys_config;
+
+    sys_config = NULL;
+    session = r->session;
+    conn = S2C(session);
+    counter = 0;
+
+    /*
+     * Read the system checkpoint information from the metadata file and save the snapshot related
+     * details of the last checkpoint for later query. This gets saved in the connection.
+     */
+
+    /* Search the metadata for the system information. */
```

Comment:
Have two distinct comment blocks isn't very common. The second one here is probably not necessary, since the function name pretty much duplicates the content of the comment.

Reply:
Removed redundant comments.